src/Core/Checkout/Customer/Validation/Constraint/CustomerVatIdentificationValidator.php

```diff
@@ -54,23 +49,29 @@ public function validate(mixed $vatIds, Constraint $constraint): void
         }
     }
 
-    private function shouldCheckVatIdFormat(CustomerVatIdentification $constraint): bool
+    private function getVatIdPattern(CustomerVatIdentification $constraint): ?string
     {
-        if ($constraint->getShouldCheck()) {
-            return true;
-        }
-
-        return (bool) $this->connection->fetchOne(
-            'SELECT check_vat_id_pattern FROM `country` WHERE id = :id',
+        $vatIdInformation = $this->connection->fetchAssociative(
+            'SELECT check_vat_id_pattern, vat_id_pattern FROM `country` WHERE id = :id',
             ['id' => Uuid::fromHexToBytes($constraint->getCountryId())]
         );
-    }
 
-    private function getVatPattern(CustomerVatIdentification $constraint): string
-    {
-        return (string) $this->connection->fetchOne(
-            'SELECT vat_id_pattern FROM `country` WHERE id = :id',
-            ['id' => Uuid::fromHexToBytes($constraint->getCountryId())]
-        );
+        if ($vatIdInformation === false) {
+            return null;
+        }
+
+        \assert(\array_key_exists('check_vat_id_pattern', $vatIdInformation));
+        \assert(\array_key_exists('vat_id_pattern', $vatIdInformation));
+
+        if (!$constraint->getShouldCheck() && !$vatIdInformation['check_vat_id_pattern']) {
```

Comment:
```suggestion
        if (!$constraint->getShouldCheck() && !($vatIdInformation['check_vat_id_pattern'] ?? null)) {
```

with that and the other suggestion below, PHPStan should no longer complain and you can remove the asserts